integration-tests/helpers.js

```diff
@@ -0,0 +1,122 @@
+'use strict'
+
+const express = require('express')
+const bodyParser = require('body-parser')
+const msgpack = require('msgpack-lite')
+const codec = msgpack.createCodec({ int64: true })
+const EventEmitter = require('events')
+const { fork } = require('child_process')
+const http = require('http')
+
+class FakeAgent extends EventEmitter {
+  constructor (port = 0) {
+    super()
+    this.port = port
+  }
+
+  async start () {
+    const app = express()
+    app.use(bodyParser.raw({ limit: Infinity, type: 'application/msgpack' }))
+    app.put('/v0.4/traces', (req, res) => {
+      if (req.body.length === 0) return res.status(200).send()
+      res.status(200).send({ rate_by_service: { 'service:,env:': 1 } })
+      this.emit('message', {
+        headers: req.headers,
+        payload: msgpack.decode(req.body, { codec })
+      })
+    })
+
+    return new Promise((resolve, reject) => {
+      const timeoutObj = setTimeout(() => {
```

Comment:
When can this happen without an error?

Reply:
if it takes more than 10 seconds to open up a listen socket. It's not likely, but it's probably the only thing that should cause a rejection here except for something throwing in here, but in that case we already get a rejection.